graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java

```diff
@@ -141,7 +148,11 @@ public JsonNode newSession(@Context ContainerRequestContext requestContext,
         try {
             Optional<Session> session = sessionCreator.create(sessionId, host, authToken);
             if (session.isPresent()) {
-                return sessionResponseFactory.forSession(session.get());
+                final SessionResponse token = sessionResponseFactory.forSession(session.get());
+                return Response.ok()
+                        .entity(token)
+                        .cookie(createSecureCookie(token, requestContext))
+                        .build();
```

Comment:
To support SSO via trusted headers, I think we need to trigger setting the cookie also when `#validateSession` is called. Also, when testing SSO, I didn't see a request from the frontend to the `#validateSession` endpoint at all so I guess the frontend needs some adjustments as well.